lib/Optimizer/CodeGen/ConvertToQIRAPI.cpp

```diff
@@ -1516,6 +1516,18 @@ struct AllocaOpPattern : public OpConversionPattern<cudaq::cc::AllocaOp> {
   }
 };
 
+struct ReturnOpPattern : public OpConversionPattern<func::ReturnOp> {
```

Comment:
This appears to be a nop (replacing an operation with a clone of itself) and is not necessary or desired. All `func` dialect operations will be handled at the appropriate time.